model/ResultsService.php

```diff
@@ -621,6 +621,29 @@ public function getStructuredVariables($resultIdentifier, $filter, $wantedTypes
         return $variablesByItem;
     }
 
+    /**
+     * Filters the complex array structure for variable classes
+     * @param array $structure
+     * @param array $filter classes to keep
+     */
+    public function filterStructuredVariables($structure, $filter)
```

Comment:
this is public function so it would be good at least to specify array type 
`public function filterStructuredVariables(array $structure,array $filter)`